src/Core/Addon/Theme/ThemeManager.php

```diff
@@ -544,17 +543,28 @@ private function importTranslationToDatabase(Theme $theme)
      * Get all default domain from catalog.
      *
      * @param string $locale
-     * @param \PrestaShopBundle\Translation\Provider\ThemeProvider $themeProvider
+     * @param string $themeName
      *
      * @return array
      */
-    private function getDefaultDomains($locale, $themeProvider)
+    private function getDefaultDomains($locale, $themeName)
     {
+        global $kernel; // sf kernel
+
+        if (!(null !== $kernel && $kernel instanceof \Symfony\Component\HttpKernel\KernelInterface)) {
+            return;
+        }
+
+        $providerFactory = $kernel->getContainer()->get('prestashop.translation.provider_factory');
+        /** @var \PrestaShopBundle\Translation\Provider\ThemeProvider $themeProvider */
+        $themeProvider = $providerFactory->build(
+            new ThemesType($themeName)
```

Comment:
Why not using `TranslationController::TYPE_THEMES`, with the `$themeName` and let the factory choice the right class to use?

Reply:
Provider factory knows which factory to call by the argument's class (which factory implements that Type). That's why I need to instantiate a specific object

... Or maybe I don't get the comment